### ISSUE TITLE
[Popover] Action list accessibility enhancements

### ISSUE THREAD

**Issue by mirualves** [Shopify/polaris-react#2187]
## Feature request summary

Add `aria-haspopup` as a prop to the popover, if the intention is to create an action list (`aria-haspopup=”menu”`).

Partially fixes: https://github.com/Shopify/polaris-ux/issues/161

## Rationale

When using a `Popover` with `Action list`, the behaviour of the component should be different from a generic popover: it should behave like a menu and support interaction in a manner that is consistent with user expectations about other menus of that type.

When navigating a menu using a keyboard, it's expected to use "up" and "down" arrow keys instead of "tab" to navigate between selections.

Generic popovers (anything else but `Action list`) should be implemented as a generic expandable/collapsible buttons and containers. In this case, it should not have `aria-haspopup` to take away ambiguity. Keep `aria-expanded`.

We should probably clarify that on the component documentation on the style guide.
cc @sadiesaurus 

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md) for more information.

**Comment by mirualves** [comment]
cc @sadiesaurus
